Add comparison operations

penzn opened the pull request.

penzn committed:
Add comparison operations

lemaitre commented:
Is there any plan to have mask types? Because those comparison instructions would definitely generate masks if they exist.
Typically, AVX512 and SVE comparisons do generate masks and not vectors.

penzn committed:
Notes on comparisons and heading clean up

Add implementation notes for comparisons operations. Clean up heading
levels in that section.

penzn commented:
Good point - we would need to balance operations returning masks and operations setting lanes. I am going to add a note documenting this. Masking ops are not forgotten - we would definitely need to evaluate them.

penzn committed:
Drive-by cleanup of markdown tags

Fix heading hierarchy and a typo in closing `summary` tag.
